libs/ledger-live-common/src/families/internet_computer/bridge/account.ts

```diff
@@ -0,0 +1,262 @@
+    } else if (totalSpent.gt(a.spendableBalance)) {
+      errors.amount = new NotEnoughBalance();
+    }
+  }
+
+  // log("debug", "[getTransactionStatus] finish fn");
+
+  return {
+    errors,
+    warnings,
+    estimatedFees,
+    amount,
+    totalSpent,
+  };
+};
+
+const estimateMaxSpendable = async ({
+  account,
+  transaction,
+}: {
+  account: AccountLike;
+  parentAccount?: Account | null | undefined;
+  transaction?: Transaction | null | undefined;
+}): Promise<BigNumber> => {
+  const balance = account.balance;
+
+  let maxSpendable = balance.minus(transaction?.fees ?? getEstimatedFees());
+
+  if (maxSpendable.isLessThan(0)) {
+    maxSpendable = new BigNumber(0);
+  }
+
+  return maxSpendable;
+};
+
+const signOperation: SignOperationFnSignature<Transaction> = ({
+  account,
+  deviceId,
+  transaction,
+}): Observable<SignOperationEvent> =>
+  withDevice(deviceId)(
+    (transport) =>
+      new Observable((o) => {
+        async function main() {
+          // log("debug", "[signOperation] start fn");
+
+          const { recipient, useAllAmount } = transaction;
+          let { amount } = transaction;
+          const { id: accountId, balance, xpub } = account;
+          const { address, derivationPath } = getAddress(account);
+          const fee = transaction.fees;
+          if (useAllAmount) amount = balance.minus(fee);
+
+          transaction = { ...transaction, amount };
+
+          try {
+            const { unsignedTxn, payloads } = await getUnsignedTransaction(
+              transaction,
+              account
+            );
+
+            o.next({
+              type: "device-signature-requested",
+            });
+
+            const { signedTxn } = await signICPTransaction({
+              unsignedTxn,
+              transport,
+              path: getPath(derivationPath),
+              payloads,
+              pubkey: xpub ?? "",
+            });
+
+            const { hash } = await getTxnMetadata(signedTxn);
+
+            o.next({
+              type: "device-signature-granted",
+            });
+
+            const operation: Operation = {
+              id: encodeOperationId(accountId, hash, "OUT"),
+              hash,
+              type: "OUT",
+              senders: [address],
+              recipients: [recipient],
+              accountId,
+              value: amount.plus(fee),
+              fee,
+              blockHash: null,
+              blockHeight: null,
+              date: new Date(),
+              extra: {
+                memo: transaction.memo,
+              },
+            };
+
+            o.next({
+              type: "signed",
+              signedOperation: {
+                operation,
+                signature: signedTxn,
+                expirationDate: getTxnExpirationDate(unsignedTxn),
+              },
+            });
+          } finally {
+            close(transport, deviceId);
+
+            // log("debug", "[signOperation] finish fn");
+          }
```

Comment:
This isn't required, you can remove the `finally` clause.

Reply:
Fixed